src/test/java/com/oscarcommerce/fw/CustomListener.java

```diff
@@ -0,0 +1,52 @@
+package com.oscarcommerce.fw;
+
+import org.openqa.selenium.WebDriver;
+import org.openqa.selenium.support.events.WebDriverListener;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class CustomListener implements WebDriverListener {
+    //    https://www.selenium.dev/selenium/docs/api/java/org/openqa/selenium/support/events/AbstractWebDriverEventListener.html
+//    https://www.selenium.dev/selenium/docs/api/java/org/openqa/selenium/support/events/EventFiringDecorator.html
+//    https://www.selenium.dev/selenium/docs/api/java/org/openqa/selenium/support/events/WebDriverListener.html#beforeQuit(org.openqa.selenium.WebDriver)
+    protected Logger logger = LoggerFactory.getLogger(CustomListener.class);
+
+    public CustomListener() {
+        super();
+    }
+
+//    @Override
```

Comment:
You can add e.g. following methods:

    @Override
    public void beforeGet(WebDriver driver, String url) {
        logger.info("About to open a page " + url);
    }

    @Override
    public void beforeClick(WebElement element) {
        logger.info("About to click on element " + element);
    }

    @Override
    public void beforeAnyCall(Object target, Method method, Object[] args) {
        logger.debug("About to call a method " + method + " in " + target + " with parameters " + args);
    }

    @Override
    public void afterAnyCall(Object target, Method method, Object[] args, Object result) {
        logger.debug("Method " + method + " called in " + target + " with parameters " + args + " returned %s" + result);
    }

    @Override
    public void onError(java.lang.Object target, java.lang.reflect.Method method, java.lang.Object[] args, java.lang.reflect.InvocationTargetException e) {
        logger.error("Exception caught!" + e.getMessage());
        logger.error(e.fillInStackTrace().toString());
    }